source/index.js

```diff
@@ -90,20 +90,23 @@ const refreshAuthorizationToken = async (refreshToken, options) => {
   return JSON.parse(body);
 };
 
-const getApplePublicKey = async () => {
+const getApplePublicKey = async (kid) => {
   const url = new URL(ENDPOINT_URL);
   url.pathname = '/auth/keys';
 
   const data = await request({ url: url.toString(), method: 'GET' });
-  const key = JSON.parse(data).keys[0];
+  const key = JSON.parse(data).keys.find(key => key.kid === kid);
```

Comment:
`request-promise-native` accepts a `json` option to automatically parse the response as json.

Can be written as the following for a small gain in readability
```js
const keys = await request({ url: url.toString(), method: 'GET', json: true });
const key = keys.find(k => k.kid === kid);
```

Reply:
Nice catch! Pull request is updated.